### PR TITLE
Support all the Rails 4's.

### PR DESCRIPTION
**Pull request by kaspth**
Unfortunately the private method logging_query_plan was removed in Rails 4,
and the query isn't done on a relation, so we can't call explain on it.

I'm a bit loathe to dig in to more private API. So I've dropped support for now.

cc @vangberg @anderslemke @mkyed

### CODE REVIEW

**Comment by kaspth** [comment]
Ready for review @anderslemke @mkyed @vangberg

File: lib/archiving/archive_table.rb [hunk 1]
```diff
@@ -24,116 +21,89 @@ def archive
...
 
-      def with_archive(query_or_options={}, options={})
-        if query_or_options.is_a?(Hash)
-          options = query_or_options
-        else
-          query = query_or_options
-        end
-
-        active = archive_select(self, "active")
+      def with_archive(query = nil, order: nil, limit: nil, offset: nil)
```
Comment (anderslemke) [inline lib/archiving/archive_table.rb]: This means that the gem won't support Rails < 4, right?

Comment (anderslemke) [inline lib/archiving/archive_table.rb]: Oh, wait. I mean Ruby < "something".

**Comment by anderslemke** [comment]
Looks good to me. :+1:

File: lib/archiving/archive_table.rb (hunk 1 above)
Comment (kaspth) [inline lib/archiving/archive_table.rb]: Yes, no less than Ruby 2.0.

**Pull request merged**